### PR TITLE
Upgrade database revision to the latest before any other operations d…

### LINKED ISSUE #1345: codalab_service.py: migrate first
Currently, deployment doesn't work when we require a migration because we try to call `create-root-user.py` then do the alembic migrate, but then we're not using the right database schema for the former.

Instead we need to do:
- If database exists, alembic upgrade
- create-root-user
- alembic stamp

### PR DESCRIPTION
**Pull request by candicegjing**
Fixed #1345

### CODE REVIEW

File: codalab_service.py [hunk 1]
```diff
@@ -586,18 +586,22 @@ def wait_rest_server(cmd):
...
                     'python3 scripts/create-root-user.py {}'.format(self.args.codalab_password)
                 )
             )
 
-            print_header('Initializing/migrating the database with alembic')
-            # The first time, we need to stamp; after that upgrade.
-            self.run_service_cmd(
-                'if [ $(alembic current | wc -l) -gt 0 ]; then echo upgrade; alembic upgrade head; else echo stamp; alembic stamp head; fi'
-            )
+            print_header('Stamping the database with alembic')
```
Comment (percyliang) [inline codalab_service.py:602]: This is only necessary when the `upgrade` condition is false, so maybe we want to add the if statement in there.

**Review by percyliang — APPROVED** [review]
Please double check that this works fine both with a new install and an actual migration (just manually roll back the DB via alembic and then run this script).

**Review by epicfaace — COMMENTED** [review]
Is it ok to change the order to `upgrade` first and then `stamp` (initially it was `stamp` and then `upgrade`)? Does the order matter?

**Comment by percyliang** [comment]
Order is crucial.  `stamp` then `upgrade` will not even perform migrations because you already stamp the current DB as the head.

**Review by percyliang — APPROVED** [review]

**Pull request merged**